### ISSUE TITLE
interpreter bug

### ISSUE THREAD

**Issue by slayoo** [gnudatalanguage/gdl#26]
(moved from https://sourceforge.net/p/gnudatalanguage/bugs/721/)
As reported on SF by Gilles:

This is a bug in two parts. First, there seems to be a misinterpretation of the meaning of () instead of [] in the last command, which should give exactly the same result as the command just above it. For some reason, GDL crashes due to a null address in the error handler. If this crash is avoided, the error becomes:

```
Expression must be a STRUCT in this context:
```

Now the crash code:

```
GDL> struct={array7:[3,3,7,7,7,5,5],z:ptr_new(/allocate_heap)}
GDL> 
GDL> zval=findgen(12)
GDL> 
GDL> pointer=ptr_new(struct,/no_copy)
GDL> 
GDL> *(*pointer).z=zval
GDL> 
GDL> print,(*(*pointer).z)[0:(*pointer).array7[3]]
      0.00000      1.00000      2.00000      3.00000      4.00000      5.00000      6.00000      7.00000
GDL> print,(*(*pointer).z)((*pointer).array7[1])
      3.00000          
GDL> print,(*(*pointer).z)((*pointer).array7(1))
Segmentation fault
```

**Comment by GillesDuvert** [comment]
incidentally, this is an after effect of ANTLR parser subtetlies that disappear in a simplified version of the parser. Simplified version that has the advantage of being very much faster also. But (nothing is ever easy) there are some side effects that need to be cured before tempting a release. I suggest to keep this on hold for the tile being.

**Comment by GillesDuvert** [comment]
solved by #1588
